elasticdl/python/worker/worker.py

```diff
@@ -304,40 +304,32 @@ def _update_local_model(self):
     def get_model(self):
         self._timing.start_record_time("get_model")
         if self._distribution_strategy != DistributionStrategy.ALLREDUCE:
-            variable_future_and_id_pairs = []
-            for ps_id, stub in enumerate(self._ps_stubs):
-                if ps_id not in self._ps_client.ps_to_parameter:
-                    continue
-                # async grpc call
+            dense_params, uninit_ps = self._ps_client.pull_dense_parameters(
+                self._model_versions_from_ps
+            )
+
+            for ps_id in uninit_ps:
+                # push variable to ps for initialization
+                parameters = [
+                    Tensor(name, self._non_embed_vars[name].numpy(), None)
+                    for name in self._ps_client.ps_to_parameter[ps_id]
+                ]
+                self._ps_client.push_dense_parameters(
+                    parameters, ps_id, self._model_versions_from_ps[ps_id]
+                )
                 req = elasticdl_pb2.PullDenseParametersRequest()
                 req.version = self._model_versions_from_ps[ps_id]
-                var_future = stub.pull_dense_parameters.future(req)
-                variable_future_and_id_pairs.append((var_future, ps_id))
-
-            for var_future, ps_id in variable_future_and_id_pairs:
-                res = var_future.result()
+                res = self._ps_stubs[ps_id].pull_dense_parameters(req)
```

Comment:
`res` is not assigned to self._non_embed_vars.

Also, put the initialization logic into ps_client may make the code cleaner.

Reply:
1. worker tries to pull dense parameters from the PS, and gets none
2. worker pushes local dense parameters to PS
3. worker pulls dense parameters from PS again, to make sure the dense parameters are initialized on the PS.

So, we need not assign the `res` to `self._non_embed_vars`. They have the same value.

Reply:
`PSClient` provides several basic client APIs for the PS, like `pull_dense_parameters`/`push_dense_parameters`. Initialization logic builds on top of these APIs. 
Right now, `self._non_embed_vars` are maintained by the worker, it's hard to move initialization logic into `PSClient`.

Reply:
Multiple workers may push dense parameters to PS and only one of them is accepted. Thus, workers need to assign the res to self._non_embed_vars as they may be different.

Reply:
Done, fix it.